CefSharp.WinForms/Internals/MovingListener.cs

```diff
@@ -0,0 +1,155 @@
+
+            // Listen for operating system messages 
+            switch (m.Msg)
+            {
+            // WM_MOVING:
+            case 0x0216:
+                movingRectangle = (Rectangle)Marshal.PtrToStructure(m.LParam, typeof(Rectangle));
+                x = movingRectangle.Left;
+                y = movingRectangle.Top;
+                care = true;
+                break;
+            // WM_MOVE
+            case 0x3:
+                x = (m.LParam.ToInt32() & 0xffff);
+                y = ((m.LParam.ToInt32() >> 16) & 0xffff);
+                care = true;
+                break;
+            }
+
+            // Only notify about movement if:
+            // * ParentForm Created
+            // * ParentControl Created
+            // * ParentForm Actually moved
+            // * Not currently moving (on the UI thread only of course)
+            // * The current WindowState is Normal.
+            //      This check is to simplify the effort here.
+            //      Other logic already handles the maximize/minimize
+            //      cases just fine.
+            // You might consider checking ParentControl.Visible and
+            // not notifying our parent control if the parent control isn't visible.
+            // However, if you do that, the non-Visible CEF tab will still
+            // have any SELECT drop downs rendering where they shouldn't.
+            if (care
+                && ParentForm != null
+                && ParentForm.IsHandleCreated
+                && ParentControl.IsHandleCreated
+                && ParentForm.WindowState == FormWindowState.Normal
+                && (ParentForm.Left != x
+                    || ParentForm.Top != y)
+                && !isMoving)
+            {
+                // .Left & .Right are negative when the window
+                // is transitioning from maximized to normal.
+                // If we are transitioning, the form will also receive
+                // a WM_SIZE which can deal with the move/size combo itself.
+                if (ParentForm.Left > 0
+                    && ParentForm.Right > 0)
+                {
+                    OnMoving();
+                    Kernel32.OutputDebugString(String.Format("Called OnMoving from control '{1:x}' to form '{0:x}'\r\n", ParentForm.Handle, ParentControl.Handle));
+                }
+                else
+                {
+                    Kernel32.OutputDebugString(String.Format("Skipped OnMoving from control '{1:x}' to form '{0:x}'\r\n", ParentForm.Handle, ParentControl.Handle));
+                }
+            }
+            else if (care)
+            {
+                Kernel32.OutputDebugString(String.Format("Skipped OnMoving from control '{1:x}' to form '{0:x}'\r\n", ParentForm.Handle, ParentControl.Handle));
+            }
+            this.DefWndProc(ref m);
+        }
+
+        protected virtual void OnMoving()
+        {
+            isMoving = true;
+            if (Moving != null)
```

Comment:
It shouldn't matter in this case though it's best practice to take a reference to the event handler before using.

See https://github.com/cefsharp/CefSharp/blob/master/CefSharp.WinForms/ChromiumWebBrowser.cs#L128 for example